pyiron_atomistics/interactive/quasi_newton.py

```diff
@@ -0,0 +1,398 @@
+
+    There are two types of regularization: Tikhonov regularization and eigenvalue softening
+    (`qn.input.use_eivenvalues = True`: eigenvalue softening, `... = False`: Tihkonov
+    regularization). In both cases, the regularization value is increased until the largest
+    displacement is smaller than `qn.input.max_displacement`.
+
+    Tikhonov regularization:
+
+    `x = (H + L)^{-1} * f`
+
+    where `x` is the displacement field, `H` is the Hessian matrix, `L` is the regularization
+    matrix and `f` is the force field. The regularization values get an opposite sign for the
+    directions along the negative eigenvalues, to make sure that the regularization indeed
+    regularizes when there is a saddle point configuration.
+
+    Eigenvalue softening:
+
+    `x = M * (d / (d^2 + L)) * M^{-1} * f`
+
+    where `M` is the eigenvector matrix, `d` are the eigenvalues and `L` is the regularization.
+    """
+
+    def __init__(self, project, job_name):
+        super().__init__(project, job_name)
+        self.__name__ = "QuasiNewton"
+        self.__version__ = None
+        self.input = Input()
+        self.output = Output(self)
+        self._interactive_interface = None
+
+    __init__.__doc__ = InteractiveWrapper.__init__.__doc__
+
+    def _run(self):
+        run_qn(
+            job=self.ref_job,
+            mode=self.input.mode,
+            ionic_steps=self.input.ionic_steps,
+            ionic_force_tolerance=self.input.ionic_force_tolerance,
+            ionic_energy_tolerance=self.input.ionic_energy_tolerance,
+            starting_h=self.input.starting_h,
+            diffusion_id=self.input.diffusion_id,
+            diffusion_direction=self.input.diffusion_direction,
+            use_eigenvalues=self.input.use_eigenvalues,
+            symmetrize=self.input.symmetrize,
+            max_displacement=self.input.max_displacement,
+        )
+        self.collect_output()
+
+    def run_static(self):
+        self.status.running = True
+        self.ref_job_initialize()
+        self._run()
+        if self.ref_job.server.run_mode.interactive:
+            self.ref_job.interactive_close()
+        self.status.collect = True
+        self.run()
+
+    run_static.__doc__ = InteractiveWrapper.run_static.__doc__
+
+    def interactive_close(self):
+        self.status.collect = True
+        if self.ref_job.server.run_mode.interactive:
+            self.ref_job.interactive_close()
+        self.run()
+
+    interactive_close.__doc__ = InteractiveWrapper.interactive_close.__doc__
+
+    def write_input(self):
+        pass
+
+    write_input.__doc__ = InteractiveWrapper.write_input.__doc__
+
+    def to_hdf(self, hdf=None, group_name=None):
+        super().to_hdf(hdf=hdf, group_name=group_name)
+
+    to_hdf.__doc__ = InteractiveWrapper.to_hdf.__doc__
+
+    def from_hdf(self, hdf=None, group_name=None):
+        super().from_hdf(hdf=hdf, group_name=group_name)
+
+    from_hdf.__doc__ = InteractiveWrapper.from_hdf.__doc__
```

Comment:
```suggestion
```
Leaving it out makes no difference, if you just call super anyway.

Reply:
Ah you're right good job :D

Reply:
`write_input` must be there though (which makes no sense to me but that how it works)